### PR TITLE
Add create shcmea for sqlalchemy datastore

### PR DESCRIPTION
**Pull request by zhu0629**
## Changes
Add create shcmea for sqlalchemy datastore

Fixes #. 

## Checklist

If this is a user-facing code change, like a bugfix or a new feature, please ensure that
you've fulfilled the following conditions (where applicable):

- [  ] You've added tests (in `tests/`) added which would fail without your patch
- [  ] You've updated the documentation (in `docs/`, in case of behavior changes or new
features)
- [ x ] You've added a new changelog entry (in `docs/versionhistory.rst`).

If this is a trivial change, like a typo fix or a code reformatting, then you can ignore
these instructions.

### Updating the changelog

If there are no entries after the last release, use `**UNRELEASED**` as the version.
If, say, your patch fixes issue #999, the entry should look like this:

`* Fix big bad boo-boo in the async scheduler (#999
<https://github.com/agronholm/apscheduler/issues/999>_; PR by @yourgithubaccount)`

If there's no issue linked, just link to your pull request instead by updating the
changelog after you've created the PR.

### CODE REVIEW

File: src/apscheduler/datastores/sqlalchemy.py [hunk 1]
```diff
@@ -231,6 +232,13 @@ async def _begin_transaction(
 
             yield conn
 
+    async def _create_schema(self, conn: Connection | AsyncConnection) -> None:
+        if not self.schema:
+            return
+
+        t = text(f"CREATE SCHEMA IF NOT EXISTS {self.schema};")
```
Comment (agronholm) [inline src/apscheduler/datastores/sqlalchemy.py:239]: If I am to merge this PR, this needs to use SQLAlchemy's own DDL: https://docs.sqlalchemy.org/en/20/core/ddl.html#sqlalchemy.schema.CreateSchema

**Comment by agronholm** [comment]
You checked the boxes for having updated the documentation and added a changelog. Can you point out to me where you've done this?

**Comment by agronholm** [comment]
Please don't add unrelated changes to this PR.

**Comment by agronholm** [comment]
I will make the necessary updates in `master` and you can then merge them here. I'll let you know when it's done.

**Comment by agronholm** [comment]
I've updated `master` now, and you can rebase your work against that.

File: docs/versionhistory.rst [hunk 2]
```diff
@@ -60,6 +60,7 @@ APScheduler, see the :doc:`migration section <migration>`.
   useful and reasonable
 - Fixed race condition in ``MongoDBDataStore`` that allowed multiple schedulers to
   acquire the same schedules at once
+- Add create `schema` for sqlalchemy datastore
```
Comment (agronholm) [inline docs/versionhistory.rst:63]:
```suggestion
- Changed ``SQLAlchemyDataStore`` to automatically create the explicitly specified schema if it's missing
```

File: src/apscheduler/datastores/sqlalchemy.py [hunk 3]
```diff
@@ -231,6 +232,12 @@ async def _begin_transaction(
 
             yield conn
 
+    async def _create_schema(self, conn: Connection | AsyncConnection) -> None:
+        if not self.schema:
+            return
+        t = schema.CreateSchema(name=self.schema, if_not_exists=True)
```
Comment (agronholm) [inline src/apscheduler/datastores/sqlalchemy.py:238]: I'd prefer that you imported this like `from sqlalchemy.schema import CreateSchema`.

File: src/apscheduler/datastores/sqlalchemy.py [hunk 4]
```diff
@@ -389,6 +396,7 @@ async def start(
         async for attempt in self._retry():
             with attempt:
                 async with self._begin_transaction() as conn:
+                    await self._create_schema(conn)
```
Comment (agronholm) [inline src/apscheduler/datastores/sqlalchemy.py:399]: Is there a reason for this code to be in its own method? Is there a use case for making it overrideable by the user?

**Review by agronholm — CHANGES_REQUESTED** [review]

**Comment by agronholm** [comment]
Please merge from master because your force pushes are now undoing the updates I made.

File: src/apscheduler/datastores/sqlalchemy.py (hunk 4 above)
Comment (agronholm) [reply src/apscheduler/datastores/sqlalchemy.py:399]: You didn't answer my question yet.

Comment (agronholm) [reply src/apscheduler/datastores/sqlalchemy.py:399]: I just need to resolve this question, and then I'm willing to merge this PR. If indeed you want to make schema creation overrideable by users, then the condition should be here, and the method should just create the schema unconditionally. But I need to understand why you made it into a separate method here.

Comment (zhu0629) [reply src/apscheduler/datastores/sqlalchemy.py:399]: I will add a param for this tmr.

**Comment by agronholm** [comment]
Could you please stop force-pushing your changes? It's causing some previous updates from `master` from being undone. There's no reason to do that, as I will squash the changes when merging anyway.

File: src/apscheduler/datastores/sqlalchemy.py (hunk 4 above)
Comment (agronholm) [reply src/apscheduler/datastores/sqlalchemy.py:399]: What parameter? I wasn't asking about any parameters. And don't resolve the conversation when my question remains unanswered.

Comment (zhu0629) [reply src/apscheduler/datastores/sqlalchemy.py:399]: I think a separate method is easier to maintain.

Comment (agronholm) [reply src/apscheduler/datastores/sqlalchemy.py:399]: No, it's just more code. The only plausible reason for keeping it in a separate method is to allow for users to override the behavior, but I don't currently see any reason to let users do that. Additionally, that would increase the API footprint which I don't want to do unless there is a good reason.

Comment (zhu0629) [reply src/apscheduler/datastores/sqlalchemy.py:399]: Can you modify it according to your suggestion？ thank you

Comment (agronholm) [reply src/apscheduler/datastores/sqlalchemy.py]: I've made the changes I wanted. I'll come back a while later to check on this and merge.

**Review by agronholm — APPROVED** [review]

**Comment by agronholm** [comment]
Thanks!

**Pull request merged**